publish_artifacts.py

```diff
@@ -5,70 +5,29 @@
 from multiprocessing import Pool
 import os
 import sys
-from time import sleep
 from typing import List
 
-import requests
-
-gh_token = os.environ["ACTIONS_RUNTIME_TOKEN"]
-
+import subprocess
 
 def upload_file(args):
     """
-    Uploads a file to our file upload service. The service is a GCP CloudRun
-    project that returns signed URLs to Google Storage objects we can upload to.
+    Uploads a file to AWS S3 using the AWS CLI.
     """
     try:
-        url, base, name = args
-
-        headers = {
-            "Authentication": f"Bearer {gh_token}",
-        }
-
-        # Obtain the signed-url for GCS using Fibonacci backoff/retries
-        for x in (1, 2, 3, 5, 0):
-            r = requests.put(url, headers=headers, allow_redirects=False)
-            if not r.ok:
-                correlation_id = r.headers.get("X-Correlation-ID", "?")
-                if not x:
-                    return (
-                        name,
-                        f"Unable to get signed url HTTP_{r.status_code}. Correlation ID: {correlation_id} - {r.text}",
-                    )
-                else:
-                    print(
-                        f"Error getting signed URL for {name}: Correlation ID: {correlation_id} HTTP_{r.status_code} - {r.text}",
-                        flush=True,
-                    )
-                    print(f"Retrying in {x} seconds", flush=True)
-                    sleep(x)
-
-        # Upload the file to the signed URL with backoff/retry logic
-        url = r.headers["location"]
+        base, name, bucket, prefix = args
         path = os.path.join(base, name)
-        for x in (1, 2, 3, 0):
-            r = requests.put(
-                url,
-                data=open(path, "rb"),
-                headers={"Content-type": "application/octet-stream"},
-            )
-            if not r.ok:
-                if not x:
-                    return (
-                        name,
-                        f"Unable to upload content HTTP_{r.status_code} - {r.text}",
-                    )
-                else:
-                    print(
-                        f"Unable to upload content for {name}: HTTP_{r.status_code} - {r.text}"
-                    )
-                    print(f"Retrying in {x} seconds")
-                    sleep(x)
-
+        key = f"{prefix}/{name}"
+        s3_uri = f"s3://{bucket}/{key}"
+
+        subprocess.run(
+            ["aws", "s3", "cp", path, s3_uri],
+            check=True,
+            capture_output=True,
+            text=True
+        )
```

Comment:
Looks like you are calling the CLI directly. I have a few questions:
* Why bother with all this python multi-threading logic? The `aws s3` command should have a way to do recursive uploads in an efficient way, right?
* This means the AWS CLI must be installed. I guess `runs-on: ubuntu-latest` guarantees this?
* How does security work here? I don't see any type of credentials getting passed to this workflow.